src/lib/button-toggle/button-toggle.ts

```diff
@@ -333,7 +333,7 @@ export class MatButtonToggleGroup implements ControlValueAccessor, OnInit, After
   /** Selects a value if there's a toggle that corresponds to it. */
   private _selectValue(value: any) {
     const correspondingOption = this._buttonToggles.find(toggle => {
-      return toggle.value != null && toggle.value === value;
```

Comment:
If we’re following how mat-select works, these changes should apply only in multiple mode. In single-selection mode null options act as a reset.

Reply:
Why only in multiple mode? Can we handle both?  Like maybe setting the value to null is a "reset" in both select and button toggle _unless_ there's an option inside that binds to null.  Then select that.

My use case is that I have a record that's either deemed a Failure, Not a Failure, Or not yet reviewed.  I'm storing this in the database just using a nullable boolean.  The following code doesn't select "Pending" when you load it up for the first time because of this "null means reset" philosophy

```html
<mat-button-toggle-group color="primary" [(ngModel)]="row.failure">
  <mat-button-toggle [value]="null">Pending Review</mat-button-toggle>
  <mat-button-toggle [value]="false">Incident</mat-button-toggle>
  <mat-button-toggle [value]="true">Failure</mat-button-toggle>
</mat-button-toggle-group>
```
Sounds like the same code wouldn't work if I changed it to a mat-select instead of using button-toggle.  I can work around this easily but it's kind of odd that you can't treat `null` as a real value, especially when we have `undefined` as an option in JS which seems more like a "reset" style of value.